Firestore/core/src/local/local_store.cc

```diff
@@ -417,6 +417,16 @@ bool LocalStore::ShouldPersistTargetData(const TargetData& new_target_data,
   int64_t time_delta = new_seconds - old_seconds;
   if (time_delta >= kResumeTokenMaxAgeSeconds) return true;
 
+  // Update the target cache if sufficient time has passed since the last
+  // LastLimboFreeSnapshotVersion
+  int64_t new_limbo_free_seconds =
+      new_target_data.last_limbo_free_snapshot_version().timestamp().seconds();
+  int64_t old_limbo_free_seconds =
+      old_target_data.last_limbo_free_snapshot_version().timestamp().seconds();
+  int64_t limbo_free_time_delta =
+      new_limbo_free_seconds - old_limbo_free_seconds;
+  if (limbo_free_time_delta >= kResumeTokenMaxAgeSeconds) return true;
```

Comment:
this sounds reasonable. For Android, this logic is placed inside `NotifyLocalViewChanges`. Arguably it makes more sense for the logic to be inside `ShouldPersistTargetData`.